### PR TITLE
Add local node to list

### PR DESCRIPTION
**Pull request by spylogsster**
Resolves https://github.com/brave/brave-browser/issues/13171

image

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
- Remove user data folder
- Start browser and check ipfs method in settings
- Local node should be in the list
- Select local node method, the node should be installed automatically

### CODE REVIEW

**Review by bbondy — CHANGES_REQUESTED** [review]
This looks good but since we don't give a privacy disclosure warning let's add text like this:
>  This protocol has different privacy considerations from HTTP(S). Learn more

Where learn more is a link to the same as the interstitial page.

**Comment by spylogsster** [comment]
> This looks good but since we don't give a privacy disclosure warning let's add text like this:
> 
> > This protocol has different privacy considerations from HTTP(S). Learn more
> 
> Where learn more is a link to the same as the interstitial page.


image

File: browser/resources/settings/brave_ipfs_page/brave_ipfs_page.js [hunk 1]
```diff
@@ -18,11 +18,22 @@ Polymer({
     WebUIListenerBehavior,
     PrefsBehavior
   ],
-
+  
+  /** 
+   * Keep it same as in IPFSResolveMethodTypes
+   * in brave\components\ipfs\ipfs_constants.h */
+  IPFSResolveMethodTypes: {
```
Comment (bbondy) [inline browser/resources/settings/brave_ipfs_page/brave_ipfs_page.js:25]: Maybe add another comment to brave\components\ipfs\ipfs_constants.h referencing this as well?

Comment (spylogsster) [reply browser/resources/settings/brave_ipfs_page/brave_ipfs_page.js:25]: done

**Review by bbondy — APPROVED** [review]

**Pull request merged**